packages/odyssey-react-mui/src/labs/TopNav/TopNav.tsx

```diff
@@ -0,0 +1,180 @@
+  alignItems: "center",
+  display: "flex",
+  marginInlineStart: "auto",
+}));
+
+const LinkContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  padding: odysseyDesignTokens.Spacing3,
+
+  "& a": {
+    color: `${odysseyDesignTokens.TypographyColorHeading} !important`,
+  },
+}));
+
+const SearchFieldContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  paddingBlock: odysseyDesignTokens.Spacing2,
+  paddingInline: odysseyDesignTokens.Spacing3,
+  width: "350px",
+}));
+
+const TopNavContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  alignItems: "center",
+  backgroundColor: odysseyDesignTokens.HueNeutral50,
+  display: "flex",
+  height: odysseyDesignTokens[TOP_NAV_HEIGHT_TOKEN],
+}));
+
+const TopNav = ({
+  AdditionalNavItemComponent,
+  helpPageHref,
+  SearchFieldComponent,
+  settingsPageHref,
+  topNavLinkItems,
+  userProfile,
+}: TopNavProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const { t } = useTranslation();
+
+  return (
+    <TopNavContainer odysseyDesignTokens={odysseyDesignTokens}>
+      {SearchFieldComponent && (
+        <SearchFieldContainer odysseyDesignTokens={odysseyDesignTokens}>
+          {SearchFieldComponent}
+        </SearchFieldContainer>
+      )}
+
+      {topNavLinkItems && <TopNavLinksList topNavLinkItems={topNavLinkItems} />}
+
+      <LinkAndProfileWrapper>
+        {(AdditionalNavItemComponent || settingsPageHref || helpPageHref) && (
+          <AdditionalLinkContainerWithBorder
+            odysseyDesignTokens={odysseyDesignTokens}
+          >
+            {AdditionalNavItemComponent && (
+              <AdditionalNavItemContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {AdditionalNavItemComponent}
+              </AdditionalNavItemContainer>
+            )}
+
+            {settingsPageHref && (
+              <LinkContainer odysseyDesignTokens={odysseyDesignTokens}>
+                <Link
+                  href={settingsPageHref}
+                  ariaLabel={t("topnav.settingsicon")}
+                >
+                  <SettingsIcon />
+                </Link>
+              </LinkContainer>
+            )}
+
+            {helpPageHref && (
+              <LinkContainer odysseyDesignTokens={odysseyDesignTokens}>
+                <Link href={helpPageHref} ariaLabel={t("topnav.helpicon")}>
+                  <QuestionCircleIcon />
+                </Link>
+              </LinkContainer>
+            )}
+          </AdditionalLinkContainerWithBorder>
+        )}
+
+        {userProfile && <UserProfile {...userProfile} />}
```

Comment:
We need to load the passed in component here instead.

Reply:
:thumbsup: